pkg/action/add_bundle_to_configs.go

```diff
@@ -0,0 +1,88 @@
+package action
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/otiai10/copy"
+	"github.com/sirupsen/logrus"
+
+	"github.com/operator-framework/operator-registry/internal/declcfg"
+	"github.com/operator-framework/operator-registry/pkg/registry"
+)
+
+type AddConfigRequest struct {
+	ConfigsDir string
+	Bundles    []BundleExtractor
+}
+
+type BundleAdder struct {
+	Logger *logrus.Entry
+}
+
+func NewBundleAdder(logger *logrus.Entry) BundleAdder {
+	return BundleAdder{
+		Logger: logger,
+	}
+}
+
+func (b BundleAdder) AddToConfig(request AddConfigRequest) error {
+	b.Logger.Infof("loading configs from directory")
+	decCfg, err := declcfg.LoadDir(request.ConfigsDir)
+	if err != nil {
+		return fmt.Errorf("error loading directory %q: %v", request.ConfigsDir, err)
+	}
+
+	model, err := declcfg.ConvertToModel(*decCfg)
+	if err != nil {
+		return fmt.Errorf("error converting configs to internal model:%v", err)
+	}
+
+	for _, bundle := range request.Bundles {
+		b, err := bundle.ExtractBundle(context.TODO())
```

Comment:
We should plumb the context in from the cobra command's `cmd.Context()` instead of using `context.TODO()`. That would mean adding a context parameter to this `AddToConfig` function.